### ISSUE TITLE
TypeError: this.QueryGenerator.addSchema is not a function

### ISSUE THREAD

**Issue by KoryNunn** [sequelize/sequelize#7880]
## What you are doing?

No idea, the throw happens inside a promise, thus, there is no useful trace for where the call originates.

I updated from v4 to v4.2.1 and ran the project.

```
Unhandled rejection TypeError: this.QueryGenerator.addSchema is not a function
    at Function.getTableName (/home/kory/dev/<project>/node_modules/sequelize/lib/model.js:1219:32)
```

Inspecting the process, `addSchema` absolutelly does not exist on `this.QueryGenerator`. This also doesn't seem to be a scoping issue, as `this` looks to be the correct value.

## What do you expect to happen?

NA

## What is actually happening?

NA

**Comment by KoryNunn** [comment]
Ah some more digging reveals it's inside sync. 

https://github.com/sequelize/sequelize/blob/5b8fe6c5a76d5fe80179114d700f386dc8ee0025/lib/model.js#L1101

**Comment by vmadman** [comment]
I'm running into a similar issue and cannot figure out where it is coming from.  This is a weird one...

```js
function dbReady( db ) {
  return db.findAll({
    where: {
      activity_id: "01f18574217741a97266697d775c5a51"
    }
  }).then( ... );
}
```

```
TypeError: this.QueryGenerator.addSchema is not a function
    at Function.getTableName (/somepath/node_modules/sequelize/lib/model.js:1233:32)
    at Function.findAll (/somepath/node_modules/sequelize/lib/model.js:1501:21)
    at dbReady ( *** )
    at processImmediate [as _immediateCallback] (timers.js:383:17)
```

I'm running the latest source from `master`, which is presumably `4.8.0`.

**Comment by vmadman** [comment]
.. ok, that's just strange.  I think it has to be my fault, but I don't know what I did wrong.  Reinstalling Sequelize made the error go away, immediately ...

```
$ yarn upgrade sequelize@github:sequelize/sequelize#master
```

I _suppose_ that means that I somehow, either directly or indirectly, modified the local `sequelize` source code in a way that caused the error. To my knowledge, I did not alter the source *except* by adding a few `console.log()` lines right there in `model.js` _after_ I started to receive the error.  So, idk ... 

Mind blowing ...
